### ISSUE TITLE
percent: normal values get rounded to 0% or 100% when mixed with huge values

### ISSUE THREAD

**Issue by CesarSancho** [r-lib/scales#128]
I am using percent from scales 0.5.0. If I apply it to a vector containing big numbers, the "normal" ones get rounded to zero, see this example:

> percent(c(0.8, 1e2))
[1] "80%"     "10,000%"
> percent(c(0.8, 1e3))
[1] "100%"     "100,000%"
> percent(c(0.8, 1e4))
[1] "0%"         "1,000,000%"

I am using r 3.4.2 on windows:
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

**Comment by dpseidel** [comment]
This is addressed in PR #77. Once it's merged, the `percent` and `percent_format` functions will have a new `accuracy` argument to help users control this rounding behavior. So stay tuned, we hope to get this merged soon!

**Comment by CesarSancho** [comment]
thank you Dana

I have made a patch for myself; I am not sure if it is worth contibuting and I would not know how to do it, so I paste it here below in case somebody find it useful. I just make sure that the minimum precision will be at least 1%, so numbers are not unnecesarily rounded. This fixes my issue without having to add parameters in the "percent" call:

```r
percent <- function (x){
  if (length(x) == 0) 
    return(character())
  #  x <- round_any(x, myPrecision(x)/100)
  x <- round_any(x, min( myPrecision(x)/100, 0.01) )
  paste0(comma(x * 100), "%")
}
```
